stdlib/builtins.pyi

```diff
@@ -59,7 +59,7 @@ from typing import (
     overload,
     runtime_checkable,
 )
-from typing_extensions import Literal
+from typing_extensions import Literal, SupportsIndex
```

Comment:
There's `_SupportIndex` defined in this file. Maybe we should change it to use the one defined in `typing_extensions`, but that can be a separate PR.